app/Services/CampaignService.php

```diff
@@ -55,12 +55,27 @@ public function find($id)
             // @TODO: change this to grab from Phoenix when Phoenix returns all info. we need!
             $campaign = $this->ashes->getCampaign($id);
 
+            // Handle cases where the ID was not found.
             if (empty($campaign['data']['id'])) {
-                info('missing_campaign_data', compact('id', 'campaign'));
+                $placeholder = [
+                    'id' => $id,
+                    'title' => 'Unknown Campaign (' . $id . ')',
+                    'staff_pick' => false,
+                    'causes' => [
+                        'primary' => [
+                            'name' => null,
+                        ],
+                    ],
+                    'uri' => null,
+                ];
+
+                $this->cache->store($id, $placeholder, 1440);
```

Comment:
Hmm I'm wondering about caching this. In theory, couldn't a signup id share the same id as a campaign id? Also does this make things much faster since it's not making any external calls or anything?

Reply:
@katiecrane yes I agree with you that a signup id could be the same as a campaign id! I'm a little confused why you bring that up here though? 

> Also does this make things much faster since it's not making any external calls or anything?

We're sending this as a placeholder and having it work the same way mainly so the page doesn't break if for some reason we get an empty or `''` as a campaign id.

Reply:
OOOOO oops I thought we were in a different file, I was thinking it was `signup_id` since you said there had been a signup missing the `campaign_id` but I was very mistaken!! Don't mind me!

Reply:
no worries I would have made the same mistake! So many 🆔 s!